From 071e0b72139952ce03f75088215d1b7954f3c59e Mon Sep 17 00:00:00 2001
From: Santiago Rossi <santiago_rho_psi@hotmail.com>
Date: Sat, 28 Apr 2012 17:21:30 -0300
Subject: [PATCH 24/40] =?UTF-8?q?4.4.5=20Cuenca=20de=20Sholazar:=20Or=C3=A1c?=
 =?UTF-8?q?ulo=20Hablalluvia=20herido?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Converted cpp script for NPC "Injured Rainspeeker Oracle" to SAI for quest "Fortunate Misunderstandings"
Added support for quest "Just following orders"
---
 ..._04_16_106_world_injured_rainspeeker_oracle.sql |   89 +++++++++++++
 src/server/scripts/Northrend/sholazar_basin.cpp    |  140 +-------------------
 2 files changed, 90 insertions(+), 139 deletions(-)
 create mode 100644 sql/updates/world/reanemu/2012_04_16_106_world_injured_rainspeeker_oracle.sql

diff --git a/sql/updates/world/reanemu/2012_04_16_106_world_injured_rainspeeker_oracle.sql b/sql/updates/world/reanemu/2012_04_16_106_world_injured_rainspeeker_oracle.sql
new file mode 100644
index 0000000..09f1dfc
--- /dev/null
+++ b/sql/updates/world/reanemu/2012_04_16_106_world_injured_rainspeeker_oracle.sql
@@ -0,0 +1,89 @@
+-- Injured Rainspeeker Oracle SAI
+SET @ORACLE := 28217;
+SET @CROCOLISK := 28325;
+UPDATE `creature_template` SET `npcflag`=0, `gossip_menu_id`=56002, `AIName`='SmartAI', `ScriptName`='' WHERE `entry`=@ORACLE;
+UPDATE `creature` SET `npcflag`=3 WHERE `guid`=101030;
+ 
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ORACLE AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@ORACLE,0,0,1,62,0,100,0,56002,0,0,0,12,@CROCOLISK,6,5000,0,0,0,8,0,0,0,5406.736816,4509.417480,-130.876526,3.529954,'Injured Rainspeeker Oracle - Gossip select - Summon Crocolisk'),
+(@ORACLE,0,1,2,61,0,100,0,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,'Injured Rainspeeker Oracle - Gossip select - Close gossip'),
+(@ORACLE,0,2,0,61,0,100,0,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Injured Rainspeeker Oracle - Gossip select - Say text'),
+(@ORACLE,0,3,0,20,0,100,0,12540,0,0,0,11,51321,0,0,0,0,0,1,0,0,0,0,0,0,0,'Injured Rainspeeker Oracle - Quest rewarded - Cast summon Shaman Vekjik'),
+(@ORACLE,0,4,0,19,0,100,0,12570,0,0,0,1,1,0,0,0,0,0,1,0,0,0,0,0,0,0,'Injured Rainspeeker Oracle - Quest accepted - Say text'),
+(@ORACLE,0,5,6,62,0,100,0,56002,1,0,0,1,2,0,0,0,0,0,1,0,0,0,0,0,0,0,'Injured Rainspeeker Oracle - Gossip select - Say text'),
+(@ORACLE,0,6,0,61,0,100,0,0,0,0,0,53,1,@ORACLE,0,12570,0,0,1,0,0,0,0,0,0,0,'Injured Rainspeeker Oracle - Gossip select - WP start'),
+(@ORACLE,0,7,8,58,0,100,0,29,0,0,0,1,3,0,0,0,0,0,1,0,0,0,0,0,0,0,'Injured Rainspeeker Oracle - WP reached - Say text'),
+(@ORACLE,0,8,9,61,0,100,0,0,0,0,0,15,12570,0,0,0,0,0,7,0,0,0,0,0,0,0,'Injured Rainspeeker Oracle - WP reached - Call area of explore event happens'),
+(@ORACLE,0,9,0,61,0,100,0,0,0,0,0,41,2000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Injured Rainspeeker Oracle - WP reached - Despawn');
+
+DELETE FROM `gossip_menu` WHERE `entry`=56002;
+INSERT INTO `gossip_menu` (`entry`,`text_id`) VALUES (56002,1); -- This part is hacky, we need conditions with SAI in order to have two different gossip texts
+
+DELETE FROM `gossip_menu_option` WHERE `menu_id`=56002;
+INSERT INTO `gossip_menu_option` (`menu_id`,`id`,`option_icon`,`option_text`,`option_id`,`npc_option_npcflag`,`action_menu_id`,`action_poi_id`,`box_coded`,`box_money`,`box_text`) VALUES
+(56002,0,0,"<Reach down and pull the Injured Rainspeak Oracle to its feet.>",1,1,0,0,0,0,''),
+(56002,1,0,"I am ready to travel to your village now.",1,1,0,0,0,0,'');
+
+DELETE FROM `conditions` WHERE `SourceGroup`=56002 AND `SourceTypeOrReferenceId`=15;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
+(15,56002,0,0,9,12540,0,0,0,0,'Only show gossip option when quest 12540 is added'),
+(15,56002,1,0,9,12570,0,0,0,0,'Only show gossip option when quest 12570 is added');
+
+DELETE FROM `script_texts` WHERE `npc_entry`=@ORACLE;
+DELETE FROM `creature_text` WHERE `entry`=@ORACLE;
+INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`comment`) VALUES
+(@ORACLE,0,0,"You hear a low guttural growl from nearby",16,0,100,0,0,0,'Injured Rainspeeker Oracle'),
+(@ORACLE,1,0,"Let me know when you ready to go, okay?",12,0,100,1,0,0,'Injured Rainspeeker Oracle'),
+(@ORACLE,2,0,"Home not far. Come!",12,0,100,1,0,0,'Injured Rainspeeker Oracle'),
+(@ORACLE,3,0,"Home time!",12,0,100,1,0,0,'Injured Rainspeeker Oracle');
+
+DELETE FROM `script_waypoint` WHERE `entry`=@ORACLE;
+DELETE FROM `waypoints` WHERE `entry`=@ORACLE;
+INSERT INTO `waypoints` (`entry`,`pointid`,`position_x`,`position_y`,`position_z`,`point_comment`) VALUES
+(@ORACLE,1,5399.96,4509.07,-131.053,''),
+(@ORACLE,2,5396.95,4514.84,-131.791,''),
+(@ORACLE,3,5395.16,4524.06,-132.335,''),
+(@ORACLE,4,5400.15,4526.84,-136.058,''),
+(@ORACLE,5,5403.53,4527.2,-138.907,''),
+(@ORACLE,6,5406.51,4527.47,-141.983,''),
+(@ORACLE,7,5409.16,4526.37,-143.902,''),
+(@ORACLE,8,5412.04,4523.05,-143.998,''),
+(@ORACLE,9,5415.23,4521.19,-143.961,''),
+(@ORACLE,10,5417.74,4519.74,-144.292,''),
+(@ORACLE,11,5421.77,4519.79,-145.36,''),
+(@ORACLE,12,5426.75,4520.53,-147.931,''),
+(@ORACLE,13,5429.06,4521.82,-148.919,''),
+(@ORACLE,14,5436.52,4534.63,-149.618,''),
+(@ORACLE,15,5441.52,4542.23,-149.367,''),
+(@ORACLE,16,5449.06,4553.47,-149.187,''),
+(@ORACLE,17,5453.5,4565.61,-147.526,''),
+(@ORACLE,18,5455.04,4578.6,-147.147,''),
+(@ORACLE,19,5462.32,4591.69,-147.738,''),
+(@ORACLE,20,5470.04,4603.04,-146.044,''),
+(@ORACLE,21,5475.93,4608.86,-143.152,''),
+(@ORACLE,22,5484.48,4613.78,-139.19,''),
+(@ORACLE,23,5489.03,4616.56,-137.796,''),
+(@ORACLE,24,5497.92,4629.89,-135.556,''),
+(@ORACLE,25,5514.48,4638.82,-136.19,''),
+(@ORACLE,26,5570,4654.5,-134.947,''),
+(@ORACLE,27,5578.32,4653.29,-136.896,''),
+(@ORACLE,28,5596.56,4642.26,-136.593,''),
+(@ORACLE,29,5634.02,4600.35,-137.291,'');
+
+-- Shaman Vekjik SAI
+SET @VEKJIK := 28327;
+UPDATE `creature_template` SET `AIName`='SmartAI', `ScriptName`='' WHERE `entry`=@VEKJIK;
+
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@VEKJIK AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@VEKJIK,0,0,0,1,0,100,1,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Shaman Vekjik - OOC - Say text'),
+(@VEKJIK,0,1,0,1,0,100,1,5000,5000,0,0,1,1,0,0,0,0,0,1,0,0,0,0,0,0,0,'Shaman Vekjik - OOC - Say text'),
+(@VEKJIK,0,2,0,1,0,100,1,8000,8000,0,0,1,2,0,0,0,0,0,1,0,0,0,0,0,0,0,'Shaman Vekjik - OOC - Say text'),
+(@VEKJIK,0,3,0,1,0,100,1,14000,14000,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Shaman Vekjik - OOC - Despawn');
+
+DELETE FROM `creature_text` WHERE `entry`=@VEKJIK;
+INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`comment`) VALUES
+(@VEKJIK,0,0,"Shaman Vekjik comes out of the underbrush nearby",16,0,100,0,0,0,'Shaman Vekjik'),
+(@VEKJIK,1,0,"You help Rainspeeker! I saw you help Rainspeeker! Traitor thing!",12,0,100,0,0,0,'Shaman Vekjik'),
+(@VEKJIK,2,0,"If you want to be friends with big-tongues, you not friends with Frenzyheart! I tell village! We no trust you anymore!",12,0,100,0,0,0,'Shaman Vekjik');
diff --git a/src/server/scripts/Northrend/sholazar_basin.cpp b/src/server/scripts/Northrend/sholazar_basin.cpp
index 93d0182..0c210cf 100644
--- a/src/server/scripts/Northrend/sholazar_basin.cpp
+++ b/src/server/scripts/Northrend/sholazar_basin.cpp
@@ -18,12 +18,11 @@
 /* ScriptData
 SDName: Sholazar_Basin
 SD%Complete: 100
-SDComment: Quest support: 12570, 12573, 12621.
+SDComment: Quest support: 12573, 12621.
 SDCategory: Sholazar_Basin
 EndScriptData */
 
 /* ContentData
-npc_injured_rainspeaker_oracle
 npc_vekjik
 avatar_of_freya
 EndContentData */
@@ -32,142 +31,6 @@ EndContentData */
 #include "ScriptedEscortAI.h"
 
 /*######
-## npc_injured_rainspeaker_oracle
-######*/
-
-#define GOSSIP_ITEM1 "I am ready to travel to your village now."
-
-enum eRainspeaker
-{
-    SAY_START_IRO                       = -1571000,
-    SAY_QUEST_ACCEPT_IRO                = -1571001,
-    SAY_END_IRO                         = -1571002,
-
-    QUEST_FORTUNATE_MISUNDERSTANDINGS   = 12570,
-    FACTION_ESCORTEE_A                  = 774,
-    FACTION_ESCORTEE_H                  = 775
-};
-
-class npc_injured_rainspeaker_oracle : public CreatureScript
-{
-public:
-    npc_injured_rainspeaker_oracle() : CreatureScript("npc_injured_rainspeaker_oracle") { }
-
-    struct npc_injured_rainspeaker_oracleAI : public npc_escortAI
-    {
-        npc_injured_rainspeaker_oracleAI(Creature* creature) : npc_escortAI(creature) { c_guid = creature->GetGUID(); }
-
-        uint64 c_guid;
-
-        void Reset()
-        {
-            me->RestoreFaction();
-            // if we will have other way to assign this to only one npc remove this part
-            if (GUID_LOPART(me->GetGUID()) != 101030)
-            {
-                me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_QUESTGIVER);
-                me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
-            }
-        }
-
-        void WaypointReached(uint32 waypointId)
-        {
-            Player* player = GetPlayerForEscort();
-            if (!player)
-                return;
-
-            switch (waypointId)
-            {
-                case 1:
-                    SetRun();
-                    break;
-                case 10:
-                case 11:
-                case 12:
-                case 13:
-                case 14:
-                case 15:
-                case 16:
-                case 17:
-                case 18:
-                    me->RemoveUnitMovementFlag(MOVEMENTFLAG_SWIMMING);
-                    me->RemoveUnitMovementFlag(MOVEMENTFLAG_FALLING);
-                    me->SetSpeed(MOVE_SWIM, 0.85f, true);
-                    me->AddUnitMovementFlag(MOVEMENTFLAG_SWIMMING | MOVEMENTFLAG_DISABLE_GRAVITY);
-                    break;
-                case 19:
-                    me->SetUnitMovementFlags(MOVEMENTFLAG_FALLING);
-                    break;
-                case 28:
-                    player->GroupEventHappens(QUEST_FORTUNATE_MISUNDERSTANDINGS, me);
-                    // me->RestoreFaction();
-                    DoScriptText(SAY_END_IRO, me);
-                    SetRun(false);
-                    break;
-            }
-        }
-
-        void JustDied(Unit* /*killer*/)
-        {
-            if (!HasEscortState(STATE_ESCORT_ESCORTING))
-                return;
-
-            if (Player* player = GetPlayerForEscort())
-            {
-                if (player->GetQuestStatus(QUEST_FORTUNATE_MISUNDERSTANDINGS) != QUEST_STATUS_COMPLETE)
-                    player->FailQuest(QUEST_FORTUNATE_MISUNDERSTANDINGS);
-            }
-        }
-    };
-
-    bool OnGossipHello(Player* player, Creature* creature)
-    {
-        if (creature->isQuestGiver())
-            player->PrepareQuestMenu(creature->GetGUID());
-
-        if (player->GetQuestStatus(QUEST_FORTUNATE_MISUNDERSTANDINGS) == QUEST_STATUS_INCOMPLETE)
-            player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, GOSSIP_ITEM1, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+1);
-
-        player->SEND_GOSSIP_MENU(player->GetGossipTextId(creature), creature->GetGUID());
-
-        return true;
-    }
-
-    bool OnGossipSelect(Player* player, Creature* creature, uint32 /*sender*/, uint32 action)
-    {
-        player->PlayerTalkClass->ClearMenus();
-        if (action == GOSSIP_ACTION_INFO_DEF+1)
-        {
-            CAST_AI(npc_escortAI, (creature->AI()))->Start(true, false, player->GetGUID());
-            CAST_AI(npc_escortAI, (creature->AI()))->SetMaxPlayerDistance(35.0f);
-            creature->SetUnitMovementFlags(MOVEMENTFLAG_FALLING);
-            DoScriptText(SAY_START_IRO, creature);
-
-            switch (player->GetTeam()){
-            case ALLIANCE:
-                creature->setFaction(FACTION_ESCORTEE_A);
-                break;
-            case HORDE:
-                creature->setFaction(FACTION_ESCORTEE_H);
-                break;
-            }
-        }
-        return true;
-    }
-
-    bool OnQuestAccept(Player* /*player*/, Creature* creature, Quest const* /*_Quest*/)
-    {
-        DoScriptText(SAY_QUEST_ACCEPT_IRO, creature);
-        return false;
-    }
-
-    CreatureAI* GetAI(Creature* creature) const
-    {
-        return new npc_injured_rainspeaker_oracleAI(creature);
-    }
-};
-
-/*######
 ## npc_vekjik
 ######*/
 
@@ -748,7 +611,6 @@ public:
 
 void AddSC_sholazar_basin()
 {
-    new npc_injured_rainspeaker_oracle();
     new npc_vekjik();
     new npc_avatar_of_freya();
     new npc_bushwhacker();
-- 
1.7.9.msysgit.0

